### PR TITLE
Fixed maximum size of disks and partitions (issue #2954)

### PR DESCRIPTION
**Pull request by SzymekkYT**
Fixed 2GB disks and partitions size limit (int limit). Its now 2TB!

### CODE REVIEW

**Comment by funcieqDEV** [comment]
Good job

**Comment by lemonekq** [comment]
why it took 6 months to be implemented :sob: 

Screenshot from 2024-05-01 23-05-16

**Comment by SzymekkYT** [comment]
everyone forgot i guess

**Comment by Kodarru** [comment]
lol

File: source/Cosmos.HAL2/BlockDevice/EBR.cs [hunk 1]
```diff
@@ -30,7 +30,7 @@ public EBR(byte[] aEBR)
             ParsePartition(aEBR, 462);
         }
 
-        protected void ParsePartition(byte[] aEBR, uint aLoc)
```
Comment (zarlo) [inline source/Cosmos.HAL2/BlockDevice/EBR.cs:33]: is this right? just asking as its the only change like this from unsigned to signed and stayed 32 bit

File: source/Cosmos.HAL2/BlockDevice/EBR.cs [hunk 2]
```diff
@@ -41,8 +41,8 @@ protected void ParsePartition(byte[] aEBR, uint aLoc)
             }
             else if (xSystemID != 0)
             {
-                uint xStartSector = BitConverter.ToUInt32(aEBR, (int)aLoc + 8);
-                uint xSectorCount = BitConverter.ToUInt32(aEBR, (int)aLoc + 12);
+                ulong xStartSector = BitConverter.ToUInt64(aEBR, aLoc + 8);
+                ulong xSectorCount = BitConverter.ToUInt64(aEBR, aLoc + 12);
```
[the remark this answers is not in the document]
Comment (SzymekkYT) [reply source/Cosmos.HAL2/BlockDevice/EBR.cs:45]: Nope.
BitConverter.ToUInt64 - Returns a 64-bit unsigned integer (thats what we want) converte from eight bytes at a specifed position in byte array.

[the remark this answers is not in the document]
Comment (SzymekkYT) [reply source/Cosmos.HAL2/BlockDevice/EBR.cs:45]: Also ToUInt32 and 64 need int as startIndex

[the remark this answers is not in the document]
Comment (Blackneeed) [reply source/Cosmos.HAL2/BlockDevice/EBR.cs:45]: Yeah, this should be UInt64 we've tested it

[the remark this answers is not in the document]
Comment (MishaProductions) [reply source/Cosmos.HAL2/BlockDevice/EBR.cs:45]: According to BitConverter.ToUInt32(aEBR, (int)aLoc + 8), BitConverter.ToUInt32(aEBR, (int)aLoc + 12); the offset difference is 4, which shows that it is a uint, not a ulong.

[the remark this answers is not in the document]
Comment (SzymekkYT) [reply source/Cosmos.HAL2/BlockDevice/EBR.cs]: All right! Everything fixed

**Review by MishaProductions — APPROVED** [review]

**Comment by MishaProductions** [comment]
Thanks for the PR!

**Comment by Blackneeed** [comment]
Why wasn't this merged tho?

**Comment by MishaProductions** [comment]
@zarlo can you please rereview this?

**Review by zarlo — APPROVED** [review]

**Pull request merged**